src/microbe_stage/systems/MicrobeVisualsSystem.cs

```diff
@@ -372,6 +372,15 @@ private void CreateOrganelleVisuals(Node3D parentNode, ref OrganelleContainer or
             {
                 tempMaterialsList[i].SetShaderParameter(tintParameterName, organelleColour);
             }
+
+            if (placedOrganelle.Upgrades?.CustomUpgradeData is FlagellumUpgrades flagellumUpgrades)
+            {
+                var flagellumLength = flagellumUpgrades.LengthFraction;
+
+                graphics.Position = new Vector3(0, 0, Constants.FLAGELLA_UPGRADE_VISUAL_OFFSET * flagellumLength);
+                graphics.Scale = new Vector3(1, 1, Constants.FLAGELLA_MAX_UPGRADE_VISUAL_LENGTH * flagellumLength
+                    + Constants.FLAGELLA_MIN_VISUAL_LENGTH);
+            }
```

Comment:
This is the wrong place. The `transform` is already calculated above in this method. You should make it so that the flagellum adjustment is applied there. (if there's a bug where the transform doesn't fully apply when graphics already exist, then that bug should be fixed separately).